pkg/api/controller.go

```diff
@@ -1103,7 +1104,22 @@ func (c *Controller) CreateRepository(w http.ResponseWriter, r *http.Request, bo
 		return
 	}
 
-	err := ensureStorageNamespaceRW(ctx, c.BlockAdapter, body.StorageNamespace)
+	parsedNs, err := url.ParseRequestURI(body.StorageNamespace)
+	if err != nil {
+		writeError(w, http.StatusBadRequest, "error parsing storage namespace. Please enter valid repository url.")
+		return
+	}
+	storageType, err := block.GetStorageType(parsedNs)
+	if err != nil {
+		writeError(w, http.StatusBadRequest, "error creating repository: storage type is not valid.")
+		return
+	}
+	if relStorageType := c.BlockAdapter.BlockstoreType(); relStorageType != storageType.BlockstoreType() {
```

Comment:
It is not clear what "rel" stands for.  can you please rename this variable? 
A suggestion: 
Use `adapterStorageType` instead of `relStorageType`, and `namespaceStorageType` instead of `storageType`

Reply:
Also, it is probably most readable to get rid of the variable and do: 
`if c.BlockAdapter.BlockstoreType() != namespaceStorageType.BlockstoreType()`

Reply:
No problem, will rename it. Also, I added the variable since I'm using it later when I'm writing the error.